### ISSUE TITLE
[BUG] Can NOT browse Firefox with HTTPS-Only mode

### ISSUE THREAD

**Issue by nakoo** [benbusby/whoogle-search#545]
**Describe the bug**

Currently Whoogle temporarily redirects `/session/` when visiting the root address for managing the session (#480). But it redirects from HTTP at first, regardless of whether `HTTPS-Only=1` is enforced.

Even though Whoogle used `upgrade-insecure-requests` CSP when `HTTPS-Only=1` is enforced, it still redirects from HTTP at first because CSP is applied after the initial load.

This behavior makes user can't browse Whoogle with HTTPS-Only Mode on Firefox. If `HTTPS-Only=1` is enforced, there should be no redirection from HTTP.

**To Reproduce**

1. Go to Firefox Settings
2. Enable HTTPS-Only mode
3. Visit Whoogle instance

![error message on https-only mode](https://user-images.githubusercontent.com/4975021/142770750-cb7fd672-da4d-4f10-b244-e0e5466a09f1.png)

**Deployment Method**
- [ ] Heroku (one-click deploy)
- [X] Docker
- [ ] `run` executable
- [ ] pip/pipx
- [ ] Other: [describe setup]

**Version of Whoogle Search**
- [X] Latest build from [source] (i.e. GitHub, Docker Hub, pip, etc)
- [ ] Version [version number]
- [ ] Not sure

**Desktop (please complete the following information):**
 - OS: 5.10.76-gentoo-r1-x86_64 GNU/Linux
 - Browser: Firefox
 - Version: 94.0.2

**Smartphone (please complete the following information):**
 - Device: [Doesn't apply]
 - OS: [Doesn't apply]
 - Browser: [Doesn't apply]
 - Version: [Doesn't apply]
 
**Additional context**

This issue is related to the prior issue. (#538) But I made a new issue since not only this bug happens in small parts like image or preference link, but happens all the parts, and this could be the issue considering the user base of Whoogle.

https://github.com/benbusby/whoogle-search/blob/0c5578937eec7f23a8921439e0e7cde1ef669271/app/routes.py#L169-L179

**Comment by benbusby** [comment]
This is kind of a confusing way to describe the issue...

TL;DR is that Flask incorrectly uses `http` as the protocol in `request.url`, so `https` needs to be enforced for instances with `HTTPS_ONLY` enabled. I'll work on a fix.
